pkgs/development/interpreters/spidermonkey/adds-Wno-deprecated-declarations-icu.patch

```diff
@@ -0,0 +1,30 @@
+From 281717cb1fe448caf359c2745bae328d57acf11a Mon Sep 17 00:00:00 2001
+From: "Kirill A. Korinsky" <kirill@korins.ky>
+Date: Fri, 1 Jul 2022 13:55:27 +0200
+Subject: [PATCH] Adds `-Wno-deprecated-declarations` to suppress icu's
+ warnings
+
+---
+ config/external/icu/defs.mozbuild | 2 ++
+ 1 file changed, 2 insertions(+)
+
+diff --git a/config/external/icu/defs.mozbuild b/config/external/icu/defs.mozbuild
+index bcb9f3b301a80..5b762827ef544 100644
+--- a/config/external/icu/defs.mozbuild
++++ b/config/external/icu/defs.mozbuild
+@@ -57,6 +57,7 @@ if CONFIG['CC_TYPE'] in ('clang', 'clang-cl'):
+         '-Wno-tautological-unsigned-enum-zero-compare',
+         '-Wno-unreachable-code-loop-increment',
+         '-Wno-unreachable-code-return',
++        '-Wno-deprecated-declarations',
```

Comment:
Isn't this warning only?

Reply:
it builds with any warning is an error => it fails.

Reply:
Let me double check it. Maybe it failed with different reason